cloudinit/net/netplan.py

```diff
@@ -263,6 +263,11 @@ def render_network_state(
             header += "\n"
 
         mode = 0o600 if features.NETPLAN_CONFIG_ROOT_READ_ONLY else 0o644
+        if os.path.exists(fpnplan):
+            current_mode = util.get_permissions(fpnplan)
+            if current_mode & mode == current_mode:
```

Comment:
This bitwise test won't apply the change to files with suid/sgid/sticky bit set. I think we should be consistent in behavior for those cases as well.

Reply:
@holmanb Thanks adapted this logic and the unit test to ensure sticky bit is preserved across updates to the file if perms are 'more strict' than cloud-init's defaults.

Reply:
I guess suid/guid/sticky shouldn't matter for a config file.

Reply:
Probably more a misconfiguration if those bits are set on a flat config file in /etc/netplan, I agree.  I could see use-cases for shared folders with group/world write perms like /tmp, but /etc/netplan is root:root 0xo644. Even the sticky bit, if set, and non-root perm on /etc/netplan/50-cloud-init.yaml will still prevent the owning non-root user from moving, deleting or modifying that file.
```
$ ls -l /etc/netplan/
total 1
-rw------T 1 chad chad 390 Feb  4 14:14 50-cloud-init.yaml

$ mv /etc/netplan/50-cloud-init.yaml  .
mv: cannot move '/etc/netplan/50-cloud-init.yaml' to './50-cloud-init.yaml': Permission denied
```
Yeah reverting this aspect of support. We want to blow away antiquated or irrelevant bits on these conf files. Not preserve them unnecessarily.